pkg/objectio/meta.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objectio
+
+import "github.com/matrixorigin/matrixone/pkg/vm/engine/tae/index"
+
+// +---------------------------------------------------------------------------------------------+
+// |                                           Header                                            |
+// +-------------+---------------+--------------+---------------+------------+-------------------+
+// | TableID(8B) | SegmentID(8B) | BlockID(8B)  | ColumnCnt(2B) | Chksum(4B) |  Reserved(33B)    |
+// +-------------+---------------+--------------+---------------+------------+-------------------+
+// |                                         ColumnMeta                                          |
+// +---------------------------------------------------------------------------------------------+
+// |                                         ColumnMeta                                          |
+// +---------------------------------------------------------------------------------------------+
+// |                                         ColumnMeta                                          |
+// +---------------------------------------------------------------------------------------------+
+// |                                         ..........                                          |
+// +---------------------------------------------------------------------------------------------+
+// Header Size = 64B
+// TableID = Table ID for Block
+// SegmentID = Segment ID
+// BlockID = Block ID
+// ColumnCnt = The number of column in the block
+// Chksum = Block metadata checksum
+// Reserved = 41 bytes reserved space
+type BlockMeta struct {
+	header *BlockHeader
+	//columns []*ColumnMeta
+}
+
+type BlockHeader struct {
+	tableId     uint64
+	segmentId   uint64
+	blockId     uint64
+	columnCount uint16
```

Comment:
This won't work.   There will be a 2 bytes padding here.   Why don't we put checksum at the end?    For example.

type BlockHeader struct {
...
...
columnCount uint16
dummy [34]byte
checksum uint32
}